src/core/ddsi/src/q_entity.c

```diff
@@ -1807,7 +1807,9 @@ static void reader_add_connection (struct reader *rd, struct proxy_writer *pwr,
       /* FIXME: for now, assume that the ports match for datasock_mc --
        't would be better to dynamically create and destroy sockets on
        an as needed basis. */
-      (void) ddsi_join_mc (rd->e.gv, rd->e.gv->mship, rd->e.gv->data_conn_mc, &m->ssm_src_loc, &m->ssm_mc_loc);
+      int ret = ddsi_join_mc (rd->e.gv, rd->e.gv->mship, rd->e.gv->data_conn_mc, &m->ssm_src_loc, &m->ssm_mc_loc);
+      if (ret < 0)
+        ELOGDISC (rd, "  unable to join\n");
```

Comment:
Not being able to join the multicast group is likely to be a big deal: you'd end up advertising to be reachable at a certain multicast address when in fact you aren't. The proper fallback would be not to advertise the multicast address for this reader, but that's a slightly more involved project. You're more than welcome to do that.

Otherwise, I think it should be at least be logged as an error (which normally goes to the log sink, i.e., stderr by default).